internal/pkg/xray/tree.go

```diff
@@ -62,15 +67,60 @@ func buildTree(traceMap map[string]*Trace, sourceTrace *Trace) (*Tree, error) {
 
 	// Find all the leaf nodes and their complete paths using DFS algorithm
 	var singlePath []*Segment
+
+	linkedTraceIds := maps.Keys(linkedTraceToSegment)
+	linkedTraceLimitExceeded := false
+
+	//get all linked traces in one call
+	if fetchLinkedTraces && len(linkedTraceIds) > 0 && *depth < MAX_TREE_DEPTH {
+		linkedTraceMap, err := opts.getTraces(ctx, opts.xrayClient, linkedTraceIds)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch linked traces %s with error: %w", linkedTraceIds, err)
+		}
+		*depth = *depth + 1
+		for linkedTraceId, linkedTrace := range linkedTraceMap {
+			linkedTree, err := buildTree(linkedTraceMap, linkedTrace, fetchLinkedTraces, ctx, opts, depth)
+			if err != nil {
+				return nil, fmt.Errorf("failed to build trace tree %s with error: %w", linkedTraceId, err)
+			}
+
+			parentSegment := linkedTraceToSegment[*linkedTree.Root.TraceId]
+			parentSegment.children = append(parentSegment.children, linkedTree.Root)
+		}
+	}
+	if *depth >= MAX_TREE_DEPTH {
```

Comment:
If the depth is exactly 5, should `linkedTraceLimitExceeded` be true?

Reply:
good catch ill change the logic so that the flag only gets set if the limit is actually exceeded